### PR TITLE
echidna, medusa: replace truffle with hardhat for test

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

relates to #156022

### CODE REVIEW

**Review by p-linnane — CHANGES_REQUESTED** [review]

File: Formula/e/echidna.rb [hunk 1]
```diff
@@ -24,7 +26,9 @@ class Echidna < Formula
   depends_on "ghc@9.2" => :build
   depends_on "haskell-stack" => :build
 
-  depends_on "truffle" => :test
+  # hardhat only support TLS node versions
```
[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/e/echidna.rb:29]: good catch :)

File: Formula/m/medusa.rb [hunk 2]
```diff
@@ -19,7 +21,9 @@ class Medusa < Formula
   end
 
   depends_on "go" => :build
-  depends_on "truffle" => :test
+  # hardhat only support TLS node versions
```
[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/m/medusa.rb:24]: agree :)

**Comment by chenrui333** [comment]
`clang: error: the clang compiler does not support '-march=nehalem'`

failed on intel builds.

**Comment by chenrui333** [comment]
closing for now.

**Pull request closed without merging**